src/workerd/server/server.c++

```diff
@@ -3087,18 +3088,47 @@ kj::Own<Server::Service> Server::makeWorker(kj::StringPtr name,
   auto observer = kj::atomicRefcounted<IsolateObserver>();
   auto limitEnforcer = kj::refcounted<NullIsolateLimitEnforcer>();
 
+  // Create the FsMap that will be used to map known file system
+  // roots to configurable locations.
+  // TODO(node-fs): This is set up to allow users to configure the "mount"
+  // points for known roots but we currently do not expose that in the
+  // config. So for now this just uses the defaults.
+  auto fsMap = kj::heap<FsMap>();
+
+  // This callback is used to resolve file system URLs to the actual
+  // underlying file system nodes.
+  auto vfsProvider = [workerFs = kj::Maybe<kj::Own<jsg::fs::VirtualFileSystem>>(), &fsMap = *fsMap,
```

Comment:
This kinda looks funky and smells in my opinion.
I think this entire thing should just be a virtual method of the API base-class that we override in both workerd and internally. 
We can add a utility method to Worker::Lock to simplify the API.
Then anyone who needs to access the filesystem can access that.

I think setting it on the jsg isolate makes for an odd design with callbacks that take objects by reference and lazily allocated fields that may as well be members.

Reply:
To piggy back off of that.
I feel like this entire feature fits better in the api directory/concept rather than in jsg.
There is some interaction with the module registry but mostly as a user rather than co-dependencies.
I think we could do with moving moving the logic out of jsg and into the api layer.

Reply:
Refactored to use `VirtualFileSystem::tryGetCurrent(js)` as the entry point and moved out of jsg.